src/components/Collections/Collection.tsx

```diff
@@ -136,10 +145,64 @@ export function Collection({
   ...containerProps
 }: { collectionId: number } & Omit<ContainerProps, 'children'>) {
   const [opened, setOpened] = useState(false);
+  const utils = trpc.useContext();
+  const user = useCurrentUser();
 
   const { data: { collection, permissions } = {}, isLoading } = trpc.collection.getById.useQuery({
     id: collectionId,
   });
+  // Using this query might be more performant all together as there is a high likelyhood
+  // that it's been preloaded by the user.
+  const { data: homeBlocks = [] } = trpc.homeBlock.getHomeBlocks.useQuery();
+  const collectionHomeBlock = useMemo(() => {
+    if (!user) {
+      return null;
+    }
+
+    return homeBlocks.find((homeBlock) => {
+      const metadata = homeBlock.metadata as HomeBlockMetaSchema;
+      return metadata.collection?.id === collectionId && homeBlock.userId === user.id;
+    });
+  }, [homeBlocks, collectionId, user]);
+
+  const createCollectionHomeBlock = trpc.homeBlock.createCollectionHomeBlock.useMutation({
+    async onSuccess() {
+      showNotification({
+        id: 'home-page-updated',
+        title: 'Home page has been updated',
+        message: `This collection has been added to your home page`,
+        color: 'teal',
+        icon: <IconCheck size={18} />,
+      });
```

Comment:
We've got a handy custom method called `showSuccessNotification` for these cases
```suggestion
      showSuccessNotification({
        title: 'Home page has been updated',
        message: `This collection has been added to your home page`,
      });
```